exec/vector/src/main/java/org/apache/drill/exec/vector/complex/MapVector.java

```diff
@@ -317,6 +317,12 @@ public Object getObject(int index) {
         if (v != null && index < v.getAccessor().getValueCount()) {
           Object value = v.getAccessor().getObject(index);
           if (value != null) {
+            if ((v.getAccessor().getObject(index) instanceof Map
+                    && ((Map) v.getAccessor().getObject(index)).size() == 0)
+                || (v.getAccessor().getObject(index) instanceof List
+                    && ((List) v.getAccessor().getObject(index)).size() == 0)) {
+              continue;
+            }
```

Comment:
Does this handle the difference beteen a missing field, and a field that exists, but contains an empty map or list? Examples:

    { "a" : { "b" : 10 }, "c" : [ 1, 2, 3 ] }  // Baseline case
    { "a" : { }, "c" : [ 4 ] }                       // Keep "a"?
    { "b" : [ 5 ] }                                     // Remove a: OK
    { "a" : { "b": 20 }, "c" : [ ] }             // Keep "c"?
    { "a" : { "b": 20 } }                           // Remove c: OK

That is, should we diffentiate between a empty map/list and a non-existent one?

The code seems to discard all empty maps & lists. Should the check actually be done in the parser to not pass along empty items? Is this possible (I'm on thin ice here in my detailed knowledge of value vectors...)